### PR TITLE
PRC-253 Gender displayed on confirmation page

### PR DESCRIPTION
**Pull request by rosariodiaferia-moj**
Ticket: [PRC-253](https://dsdmoj.atlassian.net/browse/PRC-253)

Description: Gender displayed on confirmation page

**Gender Provided:**
Gender-Provided

**Gender Not Provided:**
Gender-Not-Provided

[PRC-253]: https://dsdmoj.atlassian.net/browse/PRC-253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk [hunk 1]
```diff
@@ -15,6 +15,15 @@
             classes: 'confirm-name-value'
         }
     },
+    {
+        key: {
+            text: "Gender"
+        },
+        value: {
+            text: (contact.genderDescription | capitalizeFirstLetter) if contact.genderDescription else 'Not provided',
```
Comment (james-baker-moj) [inline server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk:23]: I'm a bit confused why we are doing the capitalise first letter. We should just use the reference data and update that if not happy with the format.

Comment (rosariodiaferia-moj) [reply server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk]: Yes, I didn't notice there was `getReferenceDescriptionForCode` that could be used from the `referenceDataService`.
I have created a method called `formattedGender` like `formattedFullName` and pass it to view. 
I have added the same logic to both confirmation and contact details

File: server/routes/contacts/add/contact-confirmation/contactConfirmationController.ts [hunk 2]
```diff
@@ -55,6 +59,18 @@ export default class ContactConfirmationController implements PageHandler {
     return res.redirect(`/prisoner/${journey.prisonerNumber}/contacts/search/${journeyId}`)
   }
 
+  private async formattedGender(contact: ContactDetails, user: Express.User) {
```
Comment (james-baker-moj) [inline server/routes/contacts/add/contact-confirmation/contactConfirmationController.ts:62]: You don't need this at all. `genderDescription` is already populated with the reference data description. Plus this is still capitalising.

**Review by james-baker-moj — CHANGES_REQUESTED** [review]

File: server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk (hunk 1 above)
Comment (james-baker-moj) [reply server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk]: hey, I think there must be some crossed wires. `genderDescription` already has the reference code description in it. There are 2 fields, `gender`, which is the reference data code, and `genderDescription` which is the reference data description.

File: server/routes/contacts/add/contact-confirmation/contactConfirmationController.ts (hunk 2 above)
Comment (rosariodiaferia-moj) [reply server/routes/contacts/add/contact-confirmation/contactConfirmationController.ts]: Changes reverted

File: server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk (hunk 1 above)
Comment (rosariodiaferia-moj) [reply server/views/pages/contacts/manage/contactConfirmation/basicDetails.njk]: gender displayed using `geneder.genderDescription` on the view

**Review by james-baker-moj — APPROVED** [review]
LGTM

**Pull request merged**